src/Omega_h_adios2.cpp

```diff
@@ -0,0 +1,563 @@
+    name = pre_name+"data";
+    write_array(io, writer, as<T>(tag)->array(), name);
+  };
+  apply_to_omega_h_types(tag->type(), std::move(f));
+}
+
+void read_tag(adios2::IO &io, adios2::Engine &reader, Mesh* mesh, 
+	      int32_t d, string &pre_name)
+{
+  std::string name = pre_name+"name";
+  adios2::Variable<std::string> bpString = io.InquireVariable<std::string>(name);
+  std::string tag_name;
+  reader.Get(bpString, tag_name);
+
+  name = pre_name+"ncomps";
+  int32_t ncomps;
+  read_value(io, reader, &ncomps, name, true);
+
+  name=pre_name+"type";
+  int8_t type;
+  read_value(io, reader, &type, name, true);
+
+  name = pre_name+"n_class_ids";
+  //TODO: read class id info for rc tag to file
+  Read<int32_t> class_ids = {};
+  int32_t n_class_ids;
+    read_value(io, reader, &n_class_ids, name, true);
+    if (n_class_ids > 0) {
+      name = pre_name+"class_ids"; 
+      read_array(io, reader, class_ids, name);
+    }
+
+  auto f = [&](auto t) {
+    using T = decltype(t);
+    Read<T> array;
+    name = pre_name+"data";
+    read_array(io, reader, array, name);
+    if(is_rc_tag(tag_name)) {
+      mesh->set_rc_from_mesh_array(d,ncomps,class_ids,tag_name,array);
+    }
+    else {
+      mesh->add_tag(d, tag_name, ncomps, array, true);
+    }
+  };
+  apply_to_omega_h_types(static_cast<Omega_h_Type>(type), std::move(f));
+
+}
+
+void write_tags(adios2::IO &io, adios2::Engine &writer, Mesh* mesh, int d, std::string pref)
+{
+  std::string name = pref+"ntags_" + to_string(d);
+  write_value(io, writer, mesh->ntags(d), name, true);
+
+  for (int32_t i = 0; i < mesh->ntags(d); ++i)
+  {
+    name = pref+"tag_" + to_string(d) + "_" + to_string(i) + "_";
+    write_tag(io, writer, mesh->get_tag(d, i), name);
+  }
+
+  name = pref+"nrctags_" + to_string(d);
+  write_value(io, writer, mesh->nrctags(d), name, true);
+
+  int32_t i=0;
+  for (const auto& rc_tag : mesh->get_rc_tags(d)) 
+  {
+    auto rc_postfix_found = ((rc_tag.get()->name()).find("_rc") != std::string::npos);
+    OMEGA_H_CHECK(rc_postfix_found);
+    const auto rc_mesh_tag = mesh->get_rc_mesh_tag_from_rc_tag(d, rc_tag.get());
+    name = pref+"rctag_" + to_string(d) + "_" + to_string(i++) + "_";
+    write_tag(io, writer, rc_mesh_tag.get(), name);
+  }
+}
+
+void read_tags(adios2::IO &io, adios2::Engine &reader, Mesh* mesh, int d, std::string pref)
+{
+  int32_t ntags;
+  std::string name = pref+"ntags_" + to_string(d);
+  read_value(io, reader, &ntags, name, true);
+
+  for (Int i = 0; i < ntags; ++i)
+  {
+    name = pref+"tag_" + to_string(d) + "_" + to_string(i) + "_";
+    read_tag(io, reader, mesh, d, name);
+  }
+
+  name = pref+"nrctags_" + to_string(d);
+  read_value(io, reader, &ntags, name, true);
+  for (Int i = 0; i < ntags; ++i)
+  {
+    name = pref+"rctags_" + to_string(d) + "_" + to_string(i);
+    read_tag(io, reader, mesh, d, name);
+  }
+}
+
+void write_pbdry(adios2::IO &io, adios2::Engine &writer, Mesh* mesh, int d, std::string pref)
+{
```

Comment:
- rename to `write_partition_boundary`, in general prefer longer, but more informative names.
- make static